sunpy/physics/transforms/differential_rotation.py

```diff
@@ -90,4 +98,352 @@ def diff_rot(ddays, latitude, rot_type='howard', frame_time='sidereal'):
+    earlier time.
+
+    Parameters
+    -----------
+    x : helio-projective x-co-ordinate in arcseconds using astropy units (can
+        be an array)
+
+    y : helio-projective y-co-ordinate in arcseconds using astropy units (can
+        be an array)
+
+    tstart : date/time to which x and y are referred; can be in any acceptable
+            time format.
+
+    tend : date/time at which x and y will be rotated to; can be
+          in any acceptable time format.
+
+    spacecraft : { None | "soho" | "stereo_a" | "stereo_b" }
+                 calculate the rotation from the point of view of the SOHO,
+                 STEREO A, or STEREO B spacecraft.
+                
+    rot_type : {'howard' | 'snodgrass' | 'allen'}
+        howard: Use values for small magnetic features from Howard et al.
+        snodgrass: Use Values from Snodgrass et. al
+        allen: Use values from Allen, Astrophysical Quantities, and simplier
+                equation.
+
+    frame_time: {'sidereal' | 'synodic'}
+        Choose 'type of day' time reference frame.
+    
+    Returns
+    -------
+    x, y : the rotated helio-projective co-ordinates (in astropy Angle
+           Quantities in units of arcseconds), with the same shape as the input
+           co-ordinates.
+
+    Examples
+    --------
+    >>> import astropy.units as u
+    >>> from sunpy.physics.transforms.differential_rotation import rot_hpc
+    >>> rot_hpc( -570 * u.arcsec, 120 * u.arcsec, '2010-09-10 12:34:56', '2010-09-10 13:34:56')
+    (<Angle -562.9105822671319 arcsec>, <Angle 119.31920621992195 arcsec>)
+
+    See Also
+    --------
+    IDL code equavalent:
+        http://hesperia.gsfc.nasa.gov/ssw/gen/idl/solar/rot_xy.pro
+
+    Note: rot_xy uses arcmin2hel.pro and hel2arcmin.pro to implement the
+    same functionality as this function.  These two functions seem to perform
+    inverse operations of each other to a high accuracy.  The corresponding
+    equivalent functions here are convert_hpc_hg and convert_hg_hpc
+    respectively. These two functions seem to perform inverse
+    operations of each other to a high accuracy.  However, the values
+    returned by arcmin2hel.pro are slightly different from those provided
+    by convert_hpc_hg.  This leads to very slightly different results from
+    rot_hpc compared to rot_xy.
+
+    TODO: the ability to do this rotation for data from the SOHO
+    point of view and the STEREO A, B point of views.
+"""
+    # must have pairs of co-ordinates
+    if np.array(x).shape != np.array(y).shape:
+        raise ValueError('Input co-ordinates must have the same shape.')
+
+    # Make sure we have enough time information to perform a solar differential
+    # rotation
+    # Start time
+    dstart = parse_time(tstart)
+    dend = parse_time(tend)
+    interval = dend - dstart
+
+    # Get the Sun's position from the vantage point at the start time
+    vstart = kwargs.get("vstart", _calc_P_B0_SD(dstart, spacecraft=spacecraft))
```

Comment:
infact why are you using `**kwargs` at all, just make this another kwarg with None as default?